plugins/obs-ffmpeg/obs-ffmpeg-vaapi.c

```diff
@@ -619,11 +761,63 @@ static bool vaapi_encode(void *data, struct encoder_frame *frame,
 				obs_av1_keyframe(packet->data, packet->size);
 		}
 		*received_packet = true;
-	} else {
-		*received_packet = false;
 	}
 
 	av_packet_unref(enc->packet);
+	return true;
+
+fail:
+	av_packet_unref(enc->packet);
+	return false;
+}
+
+static bool vaapi_encode_copy(void *data, struct encoder_frame *frame,
+			      struct encoder_packet *packet,
+			      bool *received_packet)
+{
+	struct vaapi_encoder *enc = data;
+	AVFrame *hwframe = NULL;
+	int ret;
+
+	*received_packet = false;
+
+	hwframe = av_frame_alloc();
+	if (!hwframe) {
+		warn("vaapi_encode: failed to allocate hw frame");
+		return false;
+	}
+
+	ret = av_hwframe_get_buffer(enc->vaframes_ref, hwframe, 0);
+	if (ret < 0) {
+		warn("vaapi_encode: failed to get buffer for hw frame: %s",
+		     av_err2str(ret));
+		goto fail;
+	}
+
+	copy_data(enc->vframe, frame, enc->height, enc->context->pix_fmt);
+
+	enc->vframe->pts = frame->pts;
+	hwframe->pts = frame->pts;
+	hwframe->width = enc->vframe->width;
+	hwframe->height = enc->vframe->height;
+
+	ret = av_hwframe_transfer_data(hwframe, enc->vframe, 0);
+	if (ret < 0) {
+		warn("vaapi_encode: failed to upload hw frame: %s",
+		     av_err2str(ret));
+		goto fail;
+	}
+
+	ret = av_frame_copy_props(hwframe, enc->vframe);
+	if (ret < 0) {
+		warn("vaapi_encode: failed to copy props to hw frame: %s",
```

Comment:
The function name is `vaapi_encode_copy`. I'd generally expect the log message prefix to match the function name.